pkg/utils/helper.go

```diff
@@ -232,3 +234,13 @@ func GetSourceAcctAndArn(roleARN, region, clusterName string) (string, string, e
 	sourceArn := fmt.Sprintf("arn:%s:eks:%s:%s:cluster/%s", parsedArn.Partition, region, parsedArn.AccountID, clusterName)
 	return parsedArn.AccountID, sourceArn, nil
 }
+
+// PodHasENIRequest will return true if first container of pod spec has request for eni indicating
+// it needs trunk interface from vpc-rc
+func PodHasENIRequest(pod *v1.Pod) bool {
+	if len(pod.Spec.Containers) > 0 {
+		_, hasEniRequest := pod.Spec.Containers[0].Resources.Requests[config.ResourceNamePodENI]
```

Comment:
I think we need check every container if there are more than one. It is not guaranteed always the first container is using the feature.

Reply:
ya I think thats correct

Reply:
Hi hao according to this code. If pod matched any sgp this kind of request is only added in first container. [code](https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/712887d519fd57f529371bba1ca6e7f5f49e615f/webhooks/core/pod_webhook.go#L183)

Reply:
Ohh TIL

Reply:
I see. Thanks for the pointer. I forgot webhook only mutates the first container.

Reply:
There are two moving parts now which are mutual depending... Is that possible to use a shared variable to the index that can be used at different places, such as webhook and here? In the future, if we isolate webhooks out, we may still want to point to the variable as a package reference.

Reply:
I just made it loop all containers. We can keep webhook logic and logic of finding eni request independent.